packages/vite/src/node/preview.ts

```diff
@@ -41,33 +41,48 @@ export async function preview(
   )
 
   const options = config.server || {}
-  const hostname = options.host || 'localhost'
+  let hostname: string
+  if (options.host === undefined) {
+    // Use a secure default
+    hostname = '127.0.0.1'
+  } else if (options.host === true) {
+    // The user probably passed --host in the CLI, without arguments
+    hostname = '0.0.0.0'
+  } else {
+    hostname = options.host as string
+  }
```

Comment:
Since this is reused in both files, let's extract it to a function:

```ts
const getHostname(hostnameOption: unknown): string {
  if (typeof hostnameOption === 'string') {
    return hostnameOption
  }

  // `--host` without parameters
  if (hostnameOption === true) {
    return '0.0.0.0'
  }

  // secure loopback-only default
  return '127.0.0.1'
}
```